feature-libs/organization/user-registration/components/form/user-registration-form.component.ts

```diff
@@ -0,0 +1,49 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { Country, Region } from '@spartacus/core';
+import { Title } from '@spartacus/user/profile/root';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { UserRegistrationFormService } from './user-registration-form.service';
+
+@Component({
+  selector: 'cx-user-registration-form',
+  templateUrl: './user-registration-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UserRegistrationFormComponent implements OnInit {
+  titles$: Observable<Title[]>;
+
+  countries$: Observable<Country[]>;
+
+  regions$: Observable<Region[]>;
+
+  isLoading$ = new BehaviorSubject(false);
+
+  registerForm: FormGroup;
+
+  constructor(
+    protected userRegistrationFormService: UserRegistrationFormService
+  ) {}
+
+  ngOnInit(): void {
+    this.registerForm = this.userRegistrationFormService.getForm();
+    this.titles$ = this.userRegistrationFormService.getTitles();
+    this.countries$ = this.userRegistrationFormService.getCountries();
+    this.regions$ = this.userRegistrationFormService.getRegions();
+  }
+
+  submit(): void {
+    if (this.registerForm.valid) {
+      this.isLoading$.next(true);
+
+      this.userRegistrationFormService
+        .registerUser(this.registerForm)
+        .subscribe({
+          complete: () => this.isLoading$.next(false),
+          error: () => this.isLoading$.next(false),
+        });
+    } else {
+      this.registerForm.markAllAsTouched();
+    }
```

Comment:
In fact, `isLoading$` can also be transferred to the service (+ `this.isLoading$ = this.userRegistrationFormService.isLoading$` for view), thanks to which the whole logic of this method could be transferred there

```suggestion
      this.userRegistrationFormService.registerUser(this.registerForm);       
```